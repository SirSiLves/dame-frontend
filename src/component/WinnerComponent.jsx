import React, {Component} from 'react';
import crown from "../crown.png";
import "../stylesheet/winnerPopup.css";


class WinnerComponent extends Component {

    hideWinnerPopup = () =>  {
        this.props.hideWinner();
    }


    showWinner = (winnerColor) => {

        if (winnerColor === null) return null;

        return <div className="popup" id="winnerPopup">
            <div className="popup_inner">
                <div className="d-flex flex-column bd-highlight align-items-center m-4">
                    <img className="mb-2" src={crown} alt={"queen left"} height={100} width={100}/>
                    <h3>
                        Player
                        <span className="player"> {winnerColor} </span>
                        has won the game!
                    </h3>
                </div>
            </div>
        </div>
    };


    render() {
        return (
            <div onClick={this.hideWinnerPopup}>
                {this.showWinner(this.props.winnerColor)}
            </div>
        );
    }
}

export default WinnerComponent;
