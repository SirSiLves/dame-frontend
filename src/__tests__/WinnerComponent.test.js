import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import WinnerComponent from "../component/WinnerComponent";

it('renders WinnerComponent without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<WinnerComponent />, div);
});

describe('WinnerComponent Component', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<WinnerComponent debug />);

        expect(component).toMatchSnapshot();
    });
});