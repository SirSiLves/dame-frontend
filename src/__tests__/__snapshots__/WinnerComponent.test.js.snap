// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinnerComponent Component should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
